### PR TITLE
[FIX] Switch to sniper runtime

### PR DESCRIPTION
**Pull request by Etaash-mathamsetty**
People who recently installed steam on Linux will have only the sniper runtime installed, therefore we should switch to it by default. 

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by redromnon** [comment]
Maybe another alternative is to detect if the sniper runtime is installed first and then switch to that?

For instance, my system still has the older scout runtime since I didn't use Steam for quite some time. So launching a Linux native game from Heroic will fail to launch if I enable the checkbox unless I download the runtime from Steam. But I'm pretty sure many other users may not have the same case as me.

Another way is to display a dialog asking the user to install sniper runtime in Steam.

**Comment by Etaash-mathamsetty** [comment]
> Maybe another alternative is to detect if the sniper runtime is installed first and then switch to that?
> 
> For instance, my system still has the older scout runtime since I didn't use Steam for quite some time. So launching a Linux native game from Heroic will fail to launch if I enable the checkbox unless I download the runtime from Steam. But I'm pretty sure many other users may not have the same case as me.
> 
> Another way is to display a dialog asking the user to install sniper runtime in Steam.

Ok seems fair

**Comment by flavioislima** [comment]
The Sniper runtime only works with version 8?
Perhaps we should have a method to get all available runtimes like we do for the Wine versions.

Then, depending on the Proton version we select the correct runtime. If it does not exists, simply ignore the runtime setting.

**Comment by Etaash-mathamsetty** [comment]
> The Sniper runtime only works with version 8? Perhaps we should have a method to get all available runtimes like we do for the Wine versions.
> 
> Then, depending on the Proton version we select the correct runtime. If it does not exists, simply ignore the runtime setting.

that's going to be very hard how are we supposed to detect if a certain version of proton is proton 8? And some proton-tkg builds use soldier even on the 8.x series

**Comment by imLinguin** [comment]
Doesn't tool manifest indicate that?

File: src/backend/launcher.ts [hunk 1]
```diff
@@ -121,16 +124,32 @@ async function prepareLaunch(
     gameSettings.useSteamRuntime &&
     (isNative || gameSettings.wineVersion.type === 'proton')
   if (shouldUseRuntime) {
+    let nonNativeRuntime: SteamRuntime['type'] = 'soldier'
+    if (!isNative) {
+      try {
+        const parentPath = normalize(join(gameSettings.wineVersion.bin, '..'))
+        const requiredAppId = VDF.parse(
+          readFileSync(join(parentPath, 'toolmanifest.vdf'), 'utf-8')
+        )['manifest']['require_tool_appid']
```
Comment (imLinguin) [inline src/backend/launcher.ts:133]: Why dont just use dot syntax here
```suggestion
        ).manifest?.require_tool_appid
```

**Review by imLinguin — APPROVED** [review]
Looks good, one small syntax thing to consider

File: src/backend/launcher.ts [hunk 2]
```diff
@@ -121,16 +124,32 @@ async function prepareLaunch(
...
+        if (requiredAppId === 1628350) nonNativeRuntime = 'sniper'
+      } catch (error) {
+        logError('Failed to parse toolmanifest.vdf', LogPrefix.Backend)
```
Comment (CommandMC) [inline src/backend/launcher.ts:136]: Logging the error if something goes wrong is generally a good idea. E.g.:
```suggestion
        logError(['Failed to parse toolmanifest.vdf:', error], LogPrefix.Backend)
```

File: src/backend/launcher.ts [hunk 3]
```diff
@@ -121,16 +124,32 @@ async function prepareLaunch(
...
       return {
         success: false,
         failureReason:
           'Steam Runtime is enabled, but no runtimes could be found\n' +
           `Make sure Steam ${
-            isNative ? 'is' : 'and the "SteamLinuxRuntime - Soldier" are'
+            isNative
+              ? 'is'
+              : `and the SteamLinuxRuntime - ${
+                  nonNativeRuntime === 'sniper' ? 'Sniper' : 'Solider'
```
Comment (CommandMC) [inline src/backend/launcher.ts:151]:
```suggestion
                  nonNativeRuntime === 'sniper' ? 'Sniper' : 'Soldier'
```

File: src/backend/utils.ts [hunk 4]
```diff
@@ -633,10 +633,15 @@ async function searchForExecutableOnPath(executable: string): Promise<string> {
   }
 }
 async function getSteamRuntime(
-  requestedType: 'scout' | 'soldier'
+  requestedType: 'scout' | 'sniper' | 'soldier'
```
Comment (CommandMC) [inline src/backend/utils.ts:636]: While you're touching this, might as well make this a little easier to work with:
```suggestion
  requestedType: SteamRuntime['type']
```

File: src/backend/launcher.ts [hunk 5]
```diff
@@ -121,16 +124,32 @@ async function prepareLaunch(
     gameSettings.useSteamRuntime &&
     (isNative || gameSettings.wineVersion.type === 'proton')
   if (shouldUseRuntime) {
+    let nonNativeRuntime: SteamRuntime['type'] = 'soldier'
+    if (!isNative) {
+      try {
```
Comment (CommandMC) [inline src/backend/launcher.ts:129]: A comment explaining what's being done here (and why it's done) might be helpful to future readers

**Pull request merged**